### ISSUE TITLE
Server crush (ATM9)

### ISSUE THREAD

**Issue by prejas3** [Kotori316/QuarryPlus#371]
## Describe the bug

Link to Crush Log: https://mclo.gs/JAnBRyo

## To Reproduce

1. 
2. 
3. 

If your server/client has crashed, please give me the crash report.

## Versions

* Minecraft version: [1.20.1]
* Single Player or Multi Player?: [Multi]
* Forge version: [1.20.1-47.2.20]
* QuarryPlus version: [1.20.1-1201.0.51]
* Other mods: [ATM9]

## Crash report or logs
https://mclo.gs/JAnBRyo

If your Minecraft has crashed, please attach the crash report. Please use pastebin or just paste the report.

**Comment by Thurion666** [comment]
Describe the bug
Link to Crush Log: https://pastebin.com/YTF6ngnW

To Reproduce
Steps to reproduce the behavior:
1. Playing normally eventually crashes

Versions
Minecraft version: [1.20.1]
Single Player or Multi Player?: [Multi]
Forge version: [1.20.1-47.2.20]
QuarryPlus version: [1.20.1-1201.0.51]
Other mods: [ATM9]
Crash report or logs
https://pastebin.com/YTF6ngnW

**Comment by Kotori316** [comment]
Memo

```
at com.yogpc.qp.machines.advquarry.AdvQuarryAction$CheckFluid.tick(AdvQuarryAction.java:383) ~[AdditionalEnchantedMiner-1.20.1-1201.0.51.jar%23509!/:1201.0.51] {re:classloading}
at com.yogpc.qp.machines.advquarry.TileAdvQuarry.tick(TileAdvQuarry.java:93) ~[AdditionalEnchantedMiner-1.20.1-1201.0.51.jar%23509!/:1201.0.51] {re:classloading}
at com.yogpc.qp.machines.advquarry.BlockAdvQuarry$$Lambda$75693/0x000000980810b070.m_155252_(Unknown Source) ~[?:?] {}
at com.yogpc.qp.utils.CombinedBlockEntityTicker.m_155252_(CombinedBlockEntityTicker.java:30) ~[AdditionalEnchantedMiner-1.20.1-1201.0.51.jar%23509!/:1201.0.51] {re:classloading}
```

**Comment by thedoctorjtd** [comment]
Same issue here. Have a max enchanted quarry that has been running non-stop for several days across multiple reboots without issue. XP block next to it. No changes were made to the only active quarry within at least 24-48 hours of the crash starting. After first crash server now crashes on reboot. 

https://pastebin.com/GTh4Niqg

**Comment by thedoctorjtd** [comment]
Based on loading the last working backup, it appears to have crashed after completing a 96x192 chunk quarry. Items were successfully piped out of the quarry as fast as generated, so none were remaining in the machine. ~2847 levels were in the XP block. Not sure what details are important, just trying to provide as much info as I can.

**Comment by Kotori316** [comment]
I released patched version as `v1201.0.58`

**Comment by navinman4321** [comment]
Hi I am having similar issues on 1.19.2, this thread seems closest to the issue, I had a 200*200 chunk chunkdstroyer with all the enchants.


```java
java.lang.Error: ServerHangWatchdog detected that a single server tick took 60.00 seconds (should be max 0.05)
        at com.yogpc.qp.machines.advquarry.AdvQuarryAction$CheckFluid.tick(AdvQuarryAction.java:364) ~[AdditionalEnchantedMiner-1.19.2-1192.3.25.jar%23463!/:1192.3.25] {re:classloading}
        at com.yogpc.qp.machines.advquarry.TileAdvQuarry.tick(TileAdvQuarry.java:94) ~[AdditionalEnchantedMiner-1.19.2-1192.3.25.jar%23463!/:1192.3.25] {re:classloading}
        at com.yogpc.qp.machines.advquarry.BlockAdvQuarry$$Lambda$60742/0x00007fd46dea08c8.m_155252_(Unknown Source) ~[?:?] {}
        at com.yogpc.qp.utils.CombinedBlockEntityTicker.m_155252_(CombinedBlockEntityTicker.java:30) ~[AdditionalEnchantedMiner-1.19.2-1192.3.25.jar%23463!/:1192.3.25] {re:classloading}
        at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.m_142224_(LevelChunk.java:687) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:classloading,pl:accesstransformer:B}
        at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.m_142224_(LevelChunk.java:780) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:classloading,pl:accesstransformer:B}
        at net.minecraft.world.level.Level.redirect$cao000$redirectTick(MixinLevel.java:4038) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:botania_xplat.mixins.json:LevelAccessor,pl:mixin:APP:starlight.mixins.json:common.world.LevelMixin,pl:mixin:APP:ars_nouveau.mixins.json:redstone.RedstoneLevelMixin,pl:mixin:APP:mixins.integrateddynamics.json:MixinLevel,pl:mixin:APP:utilitix.mixins.json:MixinLevel,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
        at net.minecraft.world.level.Level.m_46463_(MixinLevel.java:446) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:botania_xplat.mixins.json:LevelAccessor,pl:mixin:APP:starlight.mixins.json:common.world.LevelMixin,pl:mixin:APP:ars_nouveau.mixins.json:redstone.RedstoneLevelMixin,pl:mixin:APP:mixins.integrateddynamics.json:MixinLevel,pl:mixin:APP:utilitix.mixins.json:MixinLevel,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
        at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:331) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:sandbox:chunk_level_init,re:classloading,pl:accesstransformer:B,xf:fml:sandbox:chunk_level_init,pl:mixin:APP:cupboard.mixins.json:ServerAddEntityMixin,pl:mixin:APP:kubejs-common.mixins.json:ServerLevelMixin,pl:mixin:APP:modernfix-common.mixins.json:bugfix.chunk_deadlock.ServerLevelMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.cache_strongholds.ServerLevelMixin,pl:mixin:APP:supplementaries-common.mixins.json:ServerLevelMixin,pl:mixin:APP:botania_xplat.mixins.json:ServerLevelMixin,pl:mixin:APP:starlight.mixins.json:common.world.ServerWorldMixin,pl:mixin:APP:immersiveengineering.mixins.json:coremods.ServerWorldMixin,pl:mixin:APP:corgilib.mixins.json:MixinServerLevel,pl:mixin:APP:byg.mixins.json:common.world.MixinServerLevel,pl:mixin:APP:ad_astra-common.mixins.json:ServerLevelMixin,pl:mixin:APP:observable.common.json:ServerLevelMixin,pl:mixin:APP:betterdeserttemples.mixins.json:ServerLevelMixin,pl:mixin:APP:ars_elemental.mixins.json:ServerLevelMixin,pl:mixin:APP:create.mixins.json:accessor.ServerLevelAccessor,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:866) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
        at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:292) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:nochatreports.mixins.json:server.MixinDedicatedServer,pl:mixin:APP:tombstone.mixins.json:DedicatedServerMixin,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:806) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:654) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
        at net.minecraft.server.MinecraftServer$$Lambda$56383/0x00007fd46d936450.run(Unknown Source) ~[?:?] {}
        at java.lang.Thread.run(Thread.java:842) [?:?] {re:mixin}

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server Watchdog
Stacktrace:
        at com.yogpc.qp.machines.advquarry.AdvQuarryAction$CheckFluid.tick(AdvQuarryAction.java:364) ~[AdditionalEnchantedMiner-1.19.2-1192.3.25.jar%23463!/:1192.3.25] {re:classloading}
        at com.yogpc.qp.machines.advquarry.TileAdvQuarry.tick(TileAdvQuarry.java:94) ~[AdditionalEnchantedMiner-1.19.2-1192.3.25.jar%23463!/:1192.3.25] {re:classloading}
        at com.yogpc.qp.machines.advquarry.BlockAdvQuarry$$Lambda$60742/0x00007fd46dea08c8.m_155252_(Unknown Source) ~[?:?] {}
        at com.yogpc.qp.utils.CombinedBlockEntityTicker.m_155252_(CombinedBlockEntityTicker.java:30) ~[AdditionalEnchantedMiner-1.19.2-1192.3.25.jar%23463!/:1192.3.25] {re:classloading}
        at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.m_142224_(LevelChunk.java:687) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:classloading,pl:accesstransformer:B}
        at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.m_142224_(LevelChunk.java:780) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:classloading,pl:accesstransformer:B}
        at net.minecraft.world.level.Level.redirect$cao000$redirectTick(MixinLevel.java:4038) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:botania_xplat.mixins.json:LevelAccessor,pl:mixin:APP:starlight.mixins.json:common.world.LevelMixin,pl:mixin:APP:ars_nouveau.mixins.json:redstone.RedstoneLevelMixin,pl:mixin:APP:mixins.integrateddynamics.json:MixinLevel,pl:mixin:APP:utilitix.mixins.json:MixinLevel,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
        at net.minecraft.world.level.Level.m_46463_(MixinLevel.java:446) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:botania_xplat.mixins.json:LevelAccessor,pl:mixin:APP:starlight.mixins.json:common.world.LevelMixin,pl:mixin:APP:ars_nouveau.mixins.json:redstone.RedstoneLevelMixin,pl:mixin:APP:mixins.integrateddynamics.json:MixinLevel,pl:mixin:APP:utilitix.mixins.json:MixinLevel,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
        at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:331) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:sandbox:chunk_level_init,re:classloading,pl:accesstransformer:B,xf:fml:sandbox:chunk_level_init,pl:mixin:APP:cupboard.mixins.json:ServerAddEntityMixin,pl:mixin:APP:kubejs-common.mixins.json:ServerLevelMixin,pl:mixin:APP:modernfix-common.mixins.json:bugfix.chunk_deadlock.ServerLevelMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.cache_strongholds.ServerLevelMixin,pl:mixin:APP:supplementaries-common.mixins.json:ServerLevelMixin,pl:mixin:APP:botania_xplat.mixins.json:ServerLevelMixin,pl:mixin:APP:starlight.mixins.json:common.world.ServerWorldMixin,pl:mixin:APP:immersiveengineering.mixins.json:coremods.ServerWorldMixin,pl:mixin:APP:corgilib.mixins.json:MixinServerLevel,pl:mixin:APP:byg.mixins.json:common.world.MixinServerLevel,pl:mixin:APP:ad_astra-common.mixins.json:ServerLevelMixin,pl:mixin:APP:observable.common.json:ServerLevelMixin,pl:mixin:APP:betterdeserttemples.mixins.json:ServerLevelMixin,pl:mixin:APP:ars_elemental.mixins.json:ServerLevelMixin,pl:mixin:APP:create.mixins.json:accessor.ServerLevelAccessor,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:866) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
        at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:292) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:nochatreports.mixins.json:server.MixinDedicatedServer,pl:mixin:APP:tombstone.mixins.json:DedicatedServerMixin,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:806) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:654) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244) ~[server-1.19.2-20220805.130853-srg.jar%23819!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:modernfix-common.mixins.json:perf.dedicated_reload_executor.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
-- Thread Dump --
``` java

**Comment by Kotori316** [comment]
Released to 1.19.2
